File: reactivestreams/src/main/kotlin/reactivestreams/r2dbc/service/UserService.java
package reactivestreams.r2dbc.service;

import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import reactivestreams.r2dbc.domain.UserRepository;
import reactivestreams.r2dbc.service.response.UserResponse;
import reactor.core.publisher.Mono;

@RequiredArgsConstructor
@Service
public class UserService {

    private final UserRepository userRepository;

    public Mono<UserResponse> findById(Long userId) {
        return userRepository.findById(userId).map(UserResponse::of);
    }

}

File: reactivestreams/src/main/kotlin/reactivestreams/coroutine/service/UserService.kt
package reactivestreams.coroutine.service

import kotlinx.coroutines.reactor.awaitSingle
import org.springframework.stereotype.Service
import reactivestreams.coroutine.domain.UserRepository
import reactivestreams.coroutine.service.response.UserResponse

@Service
class UserService(
    private val userRepository: UserRepository,
) {

    suspend fun findById(userId: Long): UserResponse {
        return userRepository.findById(userId).awaitSingle().let { UserResponse(it) }
    }

}